### PR TITLE
Bug 1809609 - adds UI test page content assertion and retries

### PR DESCRIPTION
**Pull request by sv-ohorvath**
For https://mozilla-hub.atlassian.net/browse/MTE-453
The BrowserRobot verifyPageContent() method was changed during the API 30 updates, but an assertion was omitted, which is needed to validate the test result. Since this method is widely used in our tests, it is important to have it return the correct results.

Added the assertion and the retries in case of failure, to ensure the stability is not affected.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1809609

### CODE REVIEW

File: focus-android/app/src/androidTest/assets/storage_check.html [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <!DOCTYPE html>
 <html>
+<meta name="viewport" content="width=device-width">
```
Comment (AaronMT) [inline focus-android/app/src/androidTest/assets/storage_check.html:3]: Do the above changes assist with below?

File: focus-android/app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt [hunk 2]
```diff
@@ -50,7 +50,24 @@ class BrowserRobot {
...
+                    break
+                } catch (e: AssertionError) {
+                    if (i == RETRY_COUNT) {
+                        throw e
+                    } else {
+                        refreshPageIfStillLoading(expectedText)
+                    }
+                }
+            }
+        }
```
Comment (AaronMT) [inline focus-android/app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt:70]: Truthfully I'm confused between the difference between  `runWithIdleRes` and how we use other `IdlingResources` still.  ` runWithIdleRes` is doing it's own check above and `sessionLoadedIdlingResource` is doing a check [here](https://github.com/mozilla-mobile/fenix/blob/6a3adb5aeba976a7146eb96a2ef3794fae909304/app/src/androidTest/java/org/mozilla/fenix/helpers/SessionLoadedIdlingResource.kt#L30). What is the difference?

File: focus-android/app/src/androidTest/assets/storage_check.html (hunk 1 above)
Comment (sv-ohorvath) [reply focus-android/app/src/androidTest/assets/storage_check.html:3]: No, I just wanted to be able to see the text better in Firebase videos, the text was too small :)

File: focus-android/app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt (hunk 2 above)
Comment (sv-ohorvath) [reply focus-android/app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt:70]: runWithIdleRes is just registering and unregistering the resources given. In this case, the sessionIdlingRes.

**Review by AaronMT — APPROVED** [review]

**Pull request merged**